packages/common/src/Entity.ts

```diff
@@ -0,0 +1,178 @@
+import { Text } from "./Text";
+import { d3SelectionType, Widget } from "./Widget";
+
+export interface IAnnotation {
+    faChar: string;
+    tooltip?: string;
+    shape_colorFill?: string;
+    shape_colorStroke?: string;
+    image_colorFill?: string;
+    shape?: string;
+}
+
+export class Entity extends SVGWidget {
+    protected _icon_widget: Icon;
+    protected _background_widget: Shape;
+    protected _title_widget: Text;
+    protected _desc_widget: Text;
+    protected _annotation_widgets: {};
+    protected _element_anno: d3SelectionType;
+    protected _element_background: d3SelectionType;
+    protected _element_desc: d3SelectionType;
+    protected _element_icon: d3SelectionType;
+    protected _element_title: d3SelectionType;
+
+    constructor() {
+        super();
+
+        this._background_widget = new Shape();
+        this._icon_widget = new Icon();
+        this._title_widget = new Text();
+        this._desc_widget = new Text();
+        this._annotation_widgets = {};
+    }
+    enter(dn, element) {
+        super.enter.apply(this, arguments);
+        this._element_background = element.append("g").attr("class", "entity_shape");
+        this._element_icon = element.append("g").attr("class", "entity_icon");
+        this._element_title = element.append("g").attr("class", "entity_title");
+        this._element_desc = element.append("g").attr("class", "entity_desc");
+        this._element_anno = element.append("g").attr("class", "entity_anno");
+
+        this._background_widget
+            .target(this._element_background.node());
+        this._icon_widget
+            .shape_colorFill("none")
+            .shape_colorStroke("none")
+            .target(this._element_icon.node());
+        this._title_widget
+            .target(this._element_title.node());
+        this._desc_widget
+            .target(this._element_desc.node());
+    }
+    update(dn, element) {
+        super.update.apply(this, arguments);
+        this._desc_widget
+            .fontSize(this.descriptionFontSize());
+        const context = this;
+        const annotations = this._element_anno.selectAll(".annotation").data(this.annotationIcons().reverse());
+        annotations.enter().append("g")
+            .attr("class", "annotation")
+            .each(function (this: HTMLElement, _d, idx) {
+                const anno = new Icon()
+                    .diameter(context.annotationDiameter())
+                    .paddingPercent(context.annotationPaddingPercent())
+                    .target(this)
+                    .shape("square")
+                    ;
+                if (_d.faChar) anno.faChar(_d.faChar);
+                if (_d.shape) anno.shape(_d.shape);
+                if (_d.image_colorFill) anno.image_colorFill(_d.image_colorFill);
+                if (_d.shape_colorFill) anno.shape_colorFill(_d.shape_colorFill);
+                if (_d.shape_colorStroke) anno.shape_colorStroke(_d.shape_colorStroke);
```

Comment:
For all these tests you should be testing for undefined (especially the colors as color(0) should be black and not none).

Also they need to be able to unset the color for when they change from "red" to "":
```
anno.shape_colorStroke(_d.shape_colorStroke) 
```
is probably simpler and better?